Packs/IBMGuardium/Integrations/IBMSecurityGuardium/IBMSecurityGuardium.py

```diff
@@ -0,0 +1,558 @@
+    """
+    demisto.debug(f"Starting fetch_events with last_run: {last_run}, max_fetch: {max_fetch}")
+    last_fetch_time_str = last_run.get("last_fetch_time")
+    max_fetch = min(max_fetch, DEFAULT_MAX_FETCH)
+
+    # Determine fetch time range
+    now = datetime.utcnow()
+    if last_fetch_time_str:
+        # Use dateparser to handle various timestamp formats flexibly
+        last_fetch_time = dateparser.parse(last_fetch_time_str)
+        if not last_fetch_time:
+            raise DemistoException(f"Failed to parse last_fetch_time: {last_fetch_time_str}")
+    else:
+        last_fetch_time = now - timedelta(hours=1)
+        demisto.debug("No last_fetch_time found, using default: 1 hour ago")
+
+    from_date = last_fetch_time.strftime(DATE_FORMAT)
+    to_date = now.strftime(DATE_FORMAT)
+    demisto.debug(f"Fetching events from {from_date} to {to_date}")
+
+    events: list[dict[str, Any]] = []
+    offset = 0
+    field_mapping: dict[str, str] = {}
+    timestamp_field: str = ""
+    final_result = False
+    iteration = 0
+
+    while len(events) < max_fetch and not final_result and iteration < MAX_ITERATIONS:
+        iteration += 1
+        remaining = max_fetch - len(events)
+        batch_size = min(MAX_BATCH_SIZE, remaining)
+        demisto.debug(
+            f"Iteration {iteration}: Fetching batch with offset={offset}, batch_size={batch_size}, remaining={remaining}"
+        )
+
+        response = client.run_report(
+            report_id=report_id, fetch_size=batch_size, offset=offset, from_date=from_date, to_date=to_date
+        )
+
+        # Extract field mapping and timestamp field only on first batch (offset == 0)
+        if offset == 0:
+            field_mapping = extract_field_mapping(response)
+            raw_events_temp = response.get("result", {}).get("data", [])
+            if raw_events_temp:
+                first_event = map_event(raw_events_temp[0]["results"], field_mapping)
+                timestamp_field = find_timestamp_field(first_event)
+
+        raw_events = response.get("result", {}).get("data", [])
+        demisto.debug(f"Raw events count: {len(raw_events)}")
+
+        if not raw_events:
+            demisto.debug("No events returned in batch, stopping fetch loop")
+            break
+
+        # Process events
+        for raw_event in raw_events:
+            event_data = raw_event["results"]
+            mapped_event = map_event(event_data, field_mapping)
+            events.append(mapped_event)
+
+            if len(events) >= max_fetch:
+                demisto.debug(f"Reached max_fetch limit of {max_fetch}, stopping event collection")
+                break
+
+        # Check for final_result flag in the response
+        result_data = response.get("result", {})
+        final_result = result_data.get("final_result", False)
+        limit_reached = result_data.get("limit_reached", False)
```

Comment:
Lets add log when these fields are missing to detect API changes.

Reply:
Already exist: ```
demisto.debug(
            f"Batch complete: fetched {len(raw_events)} events, total so far: {len(events)}, "
            f"final_result={final_result}, limit_reached={limit_reached}"
        )
```